### DISCUSSION TITLE
Python: Pip & requirements.txt; Poetry & pyproject.toml

### DISCUSSION THREAD

**Discussion by dflock** [pkgxdev/pkgx#256] [Ideas]
Most python projects already have a file to specify dependencies. The python packaging ecosystem is pretty fragmented at the edges, but a lot of projects either use [pip](https://pip.pypa.io/en/stable/) and a [requirements.txt file](https://pip.pypa.io/en/stable/reference/requirements-file-format/), or [poetry](https://python-poetry.org/) and a [pyproject.toml file](https://python-poetry.org/docs/pyproject/).

I wouldn't want to add a duplicate dependency list in the README or a package.json file to my python projects.

**Comment by mxcl** [comment]
yeah we don't intend to duplicate this, `tea` will automatically know to setup python and pip based on these files, if you want specific versions of them then you can specify them in the README or as metadata in existing files like `requirements.txt`.

This is already specified loosely in our README since we talk about how we do magic based on your project file contents.

We just haven't added the support for python yet.

In general I approach things as “let's not make the user add yet another f-ing file”
